### PR TITLE
Add HOMEBREW_SSH_CONFIG_PATH

### PR DESCRIPTION
**Pull request by boblail**
**TL;DR: This allows specifying the path of an SSH config file that Homebrew should use instead of the default, `~/.ssh/config`, when fetching Git repos over SSH.**

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### Backstory

My company has a similar environment to the one described in https://github.com/Homebrew/brew/pull/3493.

We need to supply `SSH_AUTH_SOCK` when cloning a private tap — but we've found that there are a number of ways engineers' `~/.ssh/config` can conflict with `SSH_AUTH_SOCK`. For example, if their Config includes `IdentitiesOnly yes`, a misconfigured `IdentityAgent`, or anything malformed, `SSH_AUTH_SOCK` alone won't guarantee access.

We're looking for a better way of temporarily asserting a pristine configuration in order to set up (or repair) our engineers' environments with a script.

### CODE REVIEW

**Comment by boblail** [comment]
Thanks @MikeMcQuaid!
